src/tools/jsondocck/src/main.rs

```diff
@@ -142,6 +142,9 @@ impl CommandKind {
             (_, false) if KNOWN_DIRECTIVE_NAMES.contains(&command_name) => {
                 return None;
             }
+            (name, _) if name.starts_with("only-") => {
```

Comment:
Is this also needed given the updates to the `KNOWN_DIRECTIVE_NAMES` list?

Reply:
Yes, due to command parsing differences between the tools. The line `//@ only-aarch64-apple-darwin` otherwise causes:

```
Invalid command `//@ only-aarch`
```

Reply:
I think you should be able to fix this by editing here:

https://github.com/rust-lang/rust/blob/4ecdefdb7721124ae8b1c5eee27ed15a05f789a8/src/tools/jsondocck/src/main.rs#L157-L165

to instead be something like `(?P<cmd>[-A-Za-z]+)`, so it captures the entire command.

Reply:
I changed it to align with `htmldocck.py`, still requiring that the command not start with `-` but may include `0-9`: 

https://github.com/rust-lang/rust/blob/2205455d444f5f9da3a52e50f43c0306480218fb/src/etc/htmldocck.py#L294

That let me drop the `only-` special case here.